### ISSUE TITLE
Control timeout of instant jump/instant repeat labels separately

### ISSUE THREAD

**Issue by thenyeguy** [ggandor/lightspeed.nvim#150]
Currently, all uses of 2-character search use the same timeout (`exit_after_idle_msecs.labeled`). However, I would like the timeout to behave differently when an instant jump is triggered. Ultimately, the config I would like to use is for normal labeled jumps to have no timeout (so I have time to process the labels that have been selected), but I would like instant jumps to behave like 1 character search (e.g. the labels disappear quickly).

My general reasoning is that for a normal 2-character search, I'm not done until after I've selected the final label, and once a final label is selected all labels disappear. When using an instant repeat, there will continue to be labels even _after_ my cursor has reached its final destination, and I would like those to disappear automatically after a short timeout.

This is especially confusing for me when an instant jump is triggered, since it _feels_ like lightspeed has exited already, but there are still phantom labels in case I choose to do an instant repeat after.

I'm not sure exactly what fits best with this plugin's philosophy, but one of these two things would definitely match my intuition:

1. Add a third `exit_after_idle_msecs.instant` timeout, that I can set to some small value
2. Use `exit_after_idle_msecs.unlabeled` for instant jumps.

**Comment by ggandor** [comment]
> Currently, all uses of 2-character search use the same timeout (`exit_after_idle_msecs.labeled`).

No, that is not entirely true. It is as simple and consistent as possible: if there are labels, the `labeled` timeout value applies, if there are no labels, only match highlights, then the `unlabeled` one. The kind of motion doesn't matter. For example, if you do a repeated 2-char search, but there are lots of matches (more than the number of safe labels), you can see that after the initial `;` the labels will disappear, and change to match highlights. (Otherwise you could not exit safely.)

On the other hand, it is an intentional feature that the labels usually remain active even after `;`. You might notice that there are too many matches to `;;;;;;` through them, and want to use the label instead. What you're really asking for is a means for disabling this feature, I guess. 

The problem is, I'm actually considering _removing_ the timeout option entirely. It is a pretty useless gimmick actually: you most probably want to _do_ something after a jump; the only exception I could think of is if one has some very big `scrolloff` value, and want to use a lightspeed motion as substitute for scrolling, but that sounds a bit silly for me, and is far from being compelling enough.

**Comment by ggandor** [comment]
> otherwise it is a bug

...and indeed it is the case! I'll look into it then ASAP.

**Comment by thenyeguy** [comment]
FWIW, I discovered your simpler plugin, leap.nvim. I actually find it less disorienting for whatever reason. I think the combination of no graywash and repeat being opt-in with enter, rather than overloading the s label, an

**Comment by thenyeguy** [comment]
I discovered your leap.nvim plugin, and tried it. I actually find it less disorienting, even with `safe_labels`. I think the combination of (a) no graywash and (b) repeat being opt-in with enter means that the jumping behavior is a lot more predictable. Surprisingly I don't even miss timeouts being gone there, due to the predictability.

So maybe leap.nvim is just more what I'm looking for, and my thoughts on lightspeed are less relevant? Either way, kudos for your great plugins.
